core/src/main/java/com/tickaroo/tikxml/XmlReader.java

```diff
@@ -97,21 +98,33 @@
 
   private final BufferedSource source;
   private final Buffer buffer;
+  private final Charset charset;
   private String currentElementName;
 
-  private XmlReader(BufferedSource source) {
+  private XmlReader(BufferedSource source, Charset charset) {
     if (source == null) {
       throw new NullPointerException("source == null");
     }
     this.source = source;
     this.buffer = source.buffer();
+    this.charset = charset;
+    UNQUOTED_STRING_TERMINALS = ByteString.encodeString(" >/=\n", charset);
+    CDATA_CLOSE = ByteString.encodeString("]]>", charset);
+    CDATA_OPEN = ByteString.encodeString("<![CDATA[", charset);
+    DOCTYPE_OPEN = ByteString.encodeString("<!DOCTYPE", charset);
+    COMMENT_CLOSE = ByteString.encodeString("-->", charset);
+    XML_DECLARATION_CLOSE = ByteString.encodeString("?>", charset);
```

Comment:
I imagine these shouldn't be all caps since they are no longer static constants, same in the `XmlWriter`